### ISSUE TITLE
Handle permissions for /dev/ptp* nodes

### ISSUE THREAD

**Issue by dsseng** [systemd/systemd#31034]
### Component

udev rule files

### Is your feature request related to a problem? Please describe

We (in the PipeWire project) need to have readonly access to PTP Hardware Clocks (PHC) for our AES67 driver (pipewire-aes67 process typically runs as an ordinary user unit).

This is used to open an fd which is then converted into a clock ID for clock_gettime. No ioctls or so being used.

### Describe the solution you'd like

Udev to automatically assign a group or perhaps a global read access to PTP clock nodes so that users can use synchronized clocks. Or at least a clue whether that's a good idea.

Reference code which handles PHC ioctls and other ops on the kernel side: https://github.com/torvalds/linux/blob/master/drivers/ptp/ptp_chardev.c

### Describe alternatives you've considered

1. Currently in my test setups: udev rules like `KERNEL=="ptp[0-9]*", MODE="0644"` to allow user to open PHC for reading without extra group setup. Might actually be fine if nothing malicious can be done to a readonly clock access.
2. Use uaccess to give this access to currently logged in user.
3. Running PTP clock consumers as root. Creates a lot of permission issues because, for example, PipeWire requires communications between processes which are used to run rootless. Also security concerns.

### The systemd version you checked that didn't have the feature you are asking for

255

**Comment by poettering** [comment]
I have no idea if ptp devices are OK to open up world-accessible. if that was safe we can certainly do so. would require some confirmation from PTP kernel devs.

otherwise adding a group for this, would be fine.

**Comment by dsseng** [comment]
Thanks for your reply! Could you please reference the thread here when you redirect this to PTP subsystem maintainers?

**Comment by poettering** [comment]
uh, i have no involvement in ptp. if you want to see the change please figure out who the right people are and contact them yourself, and put the reply here, or ask them to comment here.

sorry, but i don't do ptp stuff myself, this is not a priority for myself. happy to review a patch and merge a patch but other then that don't expect me to do anything about this.

**Comment by dsseng** [comment]
okay

**Comment by dsseng** [comment]
Apparently it's not that simple. Currently under a test by Intel networking engineers and will maybe receive a kernelspace fix to ensure RO access has no side effects impacting other users and software:
- https://lists.nwtime.org/sympa/arc/linuxptp-users/2024-01/msg00040.html
- https://lists.nwtime.org/sympa/arc/linuxptp-users/2024-01/msg00042.html

**Comment by dsseng** [comment]
Yes, adding users to a group isn't a real solution. Let's wait for PTP developers to come up with how to make a truly RO access

**Comment by yuwata** [comment]
Done by af96ccfc24bc4803078a46b4ef2cdeb5decdfbcd. Closing.
